CONTRIBUTING.md

```diff
@@ -39,22 +39,63 @@ This subfolder contains some example `.rst` files that show how to implement spe
 ------
 ## Releases
 
+We use [semantic versioning](https://semver.org/). When starting a new minor "release series" like `1.11`, we create a new Git branch `1.11`. That allows us to keep making breaking changes to `main` without impacting prior releases. We can then cherry-pick relevant bug fixes from `main` to the release branch like `1.11`.
+
+The release process changes whether you are releasing the very first `rc1` for that minor release series, e.g. `1.11.0rc1` or `1.12.0rc1`. Otherwise, all other releases follow the same process.
+
+### Process for `rc1` releases
+
 1. Bump the version:
-   1. `git pull main`
-   2. Create a new branch
-   3. Bump `setup.py` and `qiskit_sphinx_theme/__init__.py` to use the new version, e.g. https://github.com/Qiskit/qiskit_sphinx_theme/pull/207
-   4. PR the change and land it
+   1. `git checkout main`
+   2. `git pull upstream main`
+   3. `git checkout -b release-<release-name>`, e.g. `release-1.11.0rc1`
+   4. Bump `setup.py` and `qiskit_sphinx_theme/__init__.py` to use the new version, e.g. https://github.com/Qiskit/qiskit_sphinx_theme/pull/207
+   5. PR the change and land it
 2. Push the Git tag:
-   1. `git pull main` to pull the version bump. If other commits have landed since the version bump, use `git revert --hard <sha>` to change to the version bump's commit (you can find the SHA with `git log`).
-   2. `git tag <version>`, e.g. 1.11.0
-   3. `git push upstream <version>`
-3. Check that the release worked:
+   1. `git checkout main`
+   2. `git pull upstream main` to pull the version bump. If other commits have landed since the version bump, use `git revert --hard <sha>` to change to the version bump's commit (you can find the SHA with `git log --oneline`).
+   3. `git tag <full-version>`, e.g. 1.11.0rc1
+   4. `git push upstream <full-version>`
+3. Create the new Git branch:
+   1. Make sure that you are still on `main` and on the commit of the version bump.
+   2. `git checkout -b <minor-release-version>`, e.g. `1.11`. This should not include the patch version.
+   3. `git push upstream <minor-release-version>`
+4. Follow the instructions in the section "Final steps shared by both processes".
+
+### Process for all other releases
+
```

Comment:
can we explicitly list out  e.g. this includes: rc2, rc3 etc, minor releases etc etc